app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/adapter/ListAdapter.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package android.support.v17.leanback.supportleanbackshowcase.app.room.adapter;
+
+import android.support.v17.leanback.widget.ObjectAdapter;
+import android.support.v17.leanback.widget.Presenter;
+import android.support.v17.leanback.widget.PresenterSelector;
+import android.support.v7.util.DiffUtil;
+import android.support.v7.util.ListUpdateCallback;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class ListAdapter<T> extends ObjectAdapter {
+
+    // For debugging purpose
+    private final static String TAG =  "ListAdapter";
+    private final static Boolean DEBUG = true;
+
+    private List<T> mItems = new ArrayList<>();
+
+    /**
+     * Constructs an adapter with the given {@link PresenterSelector}.
+     */
+    public ListAdapter(PresenterSelector presenterSelector) {
+        super(presenterSelector);
+    }
+
+    /**
+     * Constructs an adapter that uses the given {@link Presenter} for all items.
+     */
+    public ListAdapter(Presenter presenter) {
+        super(presenter);
+    }
+
+    /**
+     * Constructs an adapter.
+     */
+    public ListAdapter() {
+        super();
+    }
+
+
+    public int size() {
+        return mItems.size();
+    }
+
+    @Override
+    public T get(int position) {
+        return mItems.get(position);
+    }
+
+    /**
+     * Returns the index for the first occurrence of item in the adapter, or -1 if
+     * not found.
+     *
+     * @param item The item to find in the list.
+     * @return Index of the first occurrence of the item in the adapter, or -1
+     * if not found.
+     */
+    public int indexOf(T item) {
+        return mItems.indexOf(item);
+    }
+
+    /**
+     * Notify that the content of a range of items changed. Note that this is
+     * not same as items being added or removed.
+     *
+     * @param positionStart The position of first item that has changed.
+     * @param itemCount     The count of how many items have changed.
+     */
+    public void notifyArrayItemRangeChanged(int positionStart, int itemCount) {
```

Comment:
remove this method, just use the one in parent class.

Reply:
Sure, will do thanks

Reply:
Done